src/sentry/api/endpoints/organization_member_invite/index.py

```diff
@@ -0,0 +1,283 @@
+from django.conf import settings
+from django.db import router, transaction
+from django.db.models import Q
+from drf_spectacular.utils import extend_schema
+from rest_framework import serializers
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry import audit_log, features, ratelimits
+from sentry.api.api_owners import ApiOwner
+from sentry.api.api_publish_status import ApiPublishStatus
+from sentry.api.base import region_silo_endpoint
+from sentry.api.bases.organization import OrganizationEndpoint
+from sentry.api.bases.organizationmember import MemberAndStaffPermission
+from sentry.api.endpoints.organization_member import get_allowed_org_roles
+from sentry.api.endpoints.organization_member.utils import (
+    ERR_RATE_LIMITED,
+    ROLE_CHOICES,
+    MemberConflictValidationError,
+)
+from sentry.api.paginator import OffsetPaginator
+from sentry.api.serializers import serialize
+from sentry.api.serializers.models.organizationmemberinvite import (
+    OrganizationMemberInviteSerializer,
+)
+from sentry.models.organizationmember import OrganizationMember
+from sentry.models.organizationmemberinvite import InviteStatus, OrganizationMemberInvite
+from sentry.models.team import Team, TeamStatus
+from sentry.roles import organization_roles
+from sentry.signals import member_invited
+from sentry.users.api.parsers.email import AllowedEmailField
+from sentry.users.services.user.service import user_service
+from sentry.utils import metrics
+
+
+class OrganizationMemberInviteRequestSerializer(serializers.Serializer):
+    email = AllowedEmailField(
+        max_length=75, required=True, help_text="The email address to send the invitation to."
+    )
+    orgRole = serializers.ChoiceField(
+        choices=ROLE_CHOICES,
+        default=organization_roles.get_default().id,
+        required=False,
+        help_text="The organization-level role of the new member. Roles include:",  # choices will follow in the docs
+    )
+    teams = serializers.ListField(required=False, allow_null=False, default=[])
+    sendInvite = serializers.BooleanField(
+        required=False,
+        default=True,
+        write_only=True,
+        help_text="Whether or not to send an invite notification through email. Defaults to True.",
+    )
+    reinvite = serializers.BooleanField(
+        required=False,
+        help_text="Whether or not to re-invite a user who has already been invited to the organization. Defaults to True.",
+    )
+    regenerate = serializers.BooleanField(required=False)
```

Comment:
we should reevaluate whether we need these and what they do, i'm not too familiar off the top of my head

Reply:
ok let's remove :D